tests/testthat/test-fb_filter_species_by_traits_coverage.R

```diff
@@ -0,0 +1,114 @@
+# Fake datasets ----
+
+species_traits <- data.frame(
+  species = paste0("sp", 1:4),
+  t1      = c(NA, NA, 100, 400),
+  t2      = c(2.2, 5.0, 200, 200)
+)
+
+species_traits2 <- data.frame(
+  species = paste0("sp", 1:4),
+  t1      = c(NA, NA, NA, NA),
+  t2      = c(2.2, 5.0, 200, NA)
+)
+
+species_traits3 <- data.frame(
+  species = paste0("sp", 1:4),
+  t1      = c(NA, 2.5, 100, 400),
+  t2      = c(2.2, 2.2, 2.2, NA)
+)
+
+species_traits4 <- data.frame(
+  species = paste0("sp", 1:4),
+  t1      = c(NA, NA, 100, 400),
+  t2      = c(2.2, 5.0, NA, 200)
+)
+
+
+test_that("fb_filter_species_by_traits_coverage() errors with wrong inputs", {
+  
+  # Wrong inputs ----
+  
+  expect_error(
+    fb_filter_species_by_traits_coverage(),
+    "Argument 'species_traits' (species x traits data frame) is required",
+    fixed = TRUE
+  )
+  
+  expect_error(
+    {
+      st2 <- species_traits
+      colnames(st2) <- NULL
+      fb_filter_species_by_traits_coverage(st2)
+    },
+    "The species x traits object must have column names (trait names)",
+    fixed = TRUE
+  )
+  
+  expect_error(
+    fb_filter_species_by_traits_coverage(species_traits[ , -1, drop = FALSE]),
+    "The species x traits object must contain the 'species' column",
+    fixed = TRUE
+  )
+  
+  
+  # No numeric threshold ----
+  
+  expect_error(
+    fb_filter_species_by_traits_coverage(species_traits, threshold = "a"),
+    "Coverage threshold should be a numeric value >= 0 and <= 1",
+    fixed = TRUE
+  )
+  
+  
+  # Threshold > 1 ----
+  
+  expect_error(
+    fb_filter_species_by_traits_coverage(species_traits, threshold = 2),
+    "Coverage threshold should be a numeric value >= 0 and <= 1",
+    fixed = TRUE
+  )
+  
+  
+  # Threshold < 0 ----
+  
+  expect_error(
+    fb_filter_species_by_traits_coverage(species_traits, threshold = -1),
+    "Coverage threshold should be a numeric value >= 0 and <= 1",
+    fixed = TRUE
+  )
+  
+  
+  # Check for only NA for some species ----
+  
+  expect_message(
+    fb_filter_species_by_traits_coverage(species_traits2, threshold = 0),
+    "Some species have only NA values for all traits. Maybe you would like to remove them.",
+    fixed = TRUE
+  )
+  
+  
+  # Success ----
+  
+  expect_silent(
+    test_coverage <- fb_filter_species_by_traits_coverage(species_traits, 0)
+  )
+  
+  expect_identical(nrow(test_coverage), nrow(species_traits))
+  
+  expect_silent(
+    test_coverage <- fb_filter_species_by_traits_coverage(species_traits, 1)
+  )
+  
+  
+  # Output format ----
+  
+  expect_s3_class(test_coverage, "data.frame")
```

Comment:
Even easier test would be to test `expect_identical(test_coverage, species_traits[SELECTED_LINES,])`

Reply:
It would be difficult, because the function orders the selected rows